webapp/src/components/modals/configModal.tsx

```diff
@@ -0,0 +1,760 @@
+        getChannel(mmSiteUrl);
+        if (configIndex !== null) {
+            setSelectedTeam(existingConfig.teamName);
+            setTeamName(existingConfig.teamName);
+            setDelay(existingConfig.delayInSeconds);
+            setMessage(existingConfig.message);
+            setGuestValue(existingConfig?.includeGuests ?? '');
+            setAttachmentMessage(existingConfig?.attachmentMessage ?? []);
+        }
+    }, []);
+
+    useEffect(() => {
+        preFillActions();
+    }, [actionIndex]);
+
+    useEffect(() => {
+        setTeamNameValid(Boolean(teamName.trim()));
+
+        setTeamSelectionWarning(Boolean(teamName.trim()));
+
+        if (message.length) {
+            setMessageValid(Boolean(message[0].trim()));
+        } else {
+            setMessageValid(false);
+        }
+
+        setDelayValid(delay >= 0);
+
+        setActionTypesValueValid(Boolean(actionTypesValue));
+
+        setActionDisplayNameValid(Boolean(actionDisplayName.trim()));
+
+        setActionNameValid(Boolean(actionName.trim()));
+
+        if (actionChannelsAddedTo.length) {
+            setActionChannelsAddedToValid(Boolean(actionChannelsAddedTo[0]));
+        } else {
+            setActionChannelsAddedToValid(false);
+        }
+
+        if (actionSuccessfullMessage.length) {
+            setActionSuccessfullMessageValid(Boolean(actionSuccessfullMessage[0].trim()));
+        } else {
+            setActionSuccessfullMessageValid(false);
+        }
+    }, [teamName, delay, message, attachmentMessage, actionTypesValue, actionDisplayName, actionChannelsAddedTo, actionSuccessfullMessage, actionName]);
+
+    const handlePrimary = () => {
+        if (isActionVisible) {
+            if (actionChannelsAddedToValid && actionDisplayNameValid && actionSuccessfullMessageValid && actionTypesValueValid && actionNameValid && channelApiError === '') {
+                if (configIndex !== null) {
+                    if (actionIndex === null) {
+                        actionElement.actionDisplayName = actionDisplayName;
+                        actionElement.actionName = actionName;
+                        actionElement.actionSuccessfullMessage = actionSuccessfullMessage;
+                        actionElement.actionType = actionTypesValue;
+                        actionElement.channelsAddedTo = actionChannelsAddedTo;
+                        const actions = existingConfig?.actions;
+                        if (actions) {
+                            actions.push(actionElement);
+                            existingConfig.actions = actions;
+                        }
+                    } else {
+                        structureActions();
+                    }
+                } else if (configIndex === null) {
+                    if (actionIndex === null) {
+                        structureNewActions();
+                    } else {
+                        structureActions();
+                    }
+                }
```

Comment:
```suggestion
                if (configIndex !== null && actionIndex === null) {
                        actionElement.actionDisplayName = actionDisplayName;
                        actionElement.actionName = actionName;
                        actionElement.actionSuccessfullMessage = actionSuccessfullMessage;
                        actionElement.actionType = actionTypesValue;
                        actionElement.channelsAddedTo = actionChannelsAddedTo;
                        const actions = existingConfig?.actions;
                        if (actions) {
                            actions.push(actionElement);
                            existingConfig.actions = actions;
                        }
                } else if (configIndex === null && actionIndex === null) {
                        structureNewActions();
                } else {
                        structureActions();
                }
```
Always try to reduce the nested conditions as much as possible. Verify this on your end also once